cpp/ql/src/semmle/code/cpp/exprs/Cast.qll

```diff
@@ -840,6 +840,27 @@ class ArrayToPointerConversion extends Conversion, @array_to_pointer {
   override predicate mayBeGloballyImpure() { none() }
 }
 
+/**
+ * A node representing a temporary object created as part of an expression.
+ *
+ * This is most commonly seen in the following cases (from [class.temporary]/2):
+ * - when binding a reference to a prvalue
+ * - when performing member access on a class prvalue
+ * - when performing an array-to-pointer conversion or subscripting on an array prvalue
+ * - when initializing an object of type std::initializer_list from a braced-init-list
+ * - for certain unevaluated operands
+ * - when a prvalue that has type other than cv void appears as a discarded-value expression
+ *
+ * This node will only exist if the object is of class type, and even then only if either the
+ * object's initialization or destruction is non-trivial.
```

Comment:
The GNU binary ternary operator also seems to create these expressions. Is that intended? Looking at `question_mark_colon/cfg.expected`, it looks like it's created even for trivial types.